android/pytorch_android/neural_machine_translation/src/main/java/ai/djl/examples/neuralmachinetranslation/NeuralModel.java

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+
+package ai.djl.examples.neuralmachinetranslation;
+
+import com.google.gson.internal.LinkedTreeMap;
+
+import java.io.IOException;
+import java.nio.FloatBuffer;
+import java.util.ArrayList;
+import java.util.Locale;
+
+import ai.djl.ModelException;
+import ai.djl.ndarray.NDArray;
+import ai.djl.ndarray.NDList;
+import ai.djl.ndarray.NDManager;
+import ai.djl.ndarray.types.Shape;
+import ai.djl.nn.Block;
+import ai.djl.repository.zoo.Criteria;
+import ai.djl.repository.zoo.ZooModel;
+import ai.djl.training.ParameterStore;
+
+final class NeuralModel {
+
+    private static final int HIDDEN_SIZE = 256;
+    private static final int EOS_TOKEN = 1;
+    private static final int MAX_LENGTH = 50;
+
+    private NeuralModel() {
+    }
+
+    public static ZooModel<NDList, NDList> loadModelEncoder() throws ModelException, IOException {
+        String url =
+                "https://resources.djl.ai/demo/pytorch/android/neural_machine_translation/optimized_encoder_150k.zip";
+
+        Criteria<NDList, NDList> criteria =
+                Criteria.builder()
+                        .setTypes(NDList.class, NDList.class)
+                        .optModelUrls(url)
+                        .optModelName("optimized_encoder_150k.ptl")
+                        .optEngine("PyTorch")
+                        .build();
+        return criteria.loadModel();
+    }
+
+    public static ZooModel<NDList, NDList> loadModelDecoder() throws ModelException, IOException {
+        String url =
+                "https://resources.djl.ai/demo/pytorch/android/neural_machine_translation/optimized_decoder_150k.zip";
+
+        Criteria<NDList, NDList> criteria =
+                Criteria.builder()
+                        .setTypes(NDList.class, NDList.class)
+                        .optModelUrls(url)
+                        .optModelName("optimized_decoder_150k.ptl")
+                        .optEngine("PyTorch")
+                        .build();
+        return criteria.loadModel();
+    }
+
+    public static NDList predictEncoder(String text, ZooModel<NDList, NDList> model,
+                                        LinkedTreeMap<String, Long> wrd2idx, NDManager manager)
+            throws ModelException, IOException {
+        // maps french input to id's from french file
+        String[] french = text.split(" ");
```

Comment:
This doesn't handle punctuations. Please take a look this example:
https://github.com/deepjavalibrary/djl/blob/master/examples/src/main/java/ai/djl/examples/training/TrainSentimentAnalysis.java#L69-L73